core/CronArchive.php

```diff
@@ -561,663 +418,312 @@ public function run()
         $this->logger->info($timer->__toString());
     }
 
-    public function getErrors()
+    private function isDoneFlagForPlugin($doneFlag)
     {
-    	return $this->errors;
+        return strpos($doneFlag, '.') !== false;
     }
 
-    /**
-     * End of the script
-     */
-    public function end()
+    private function archiveArrayContainsArchive($archiveArray, $archive)
     {
-        /**
-         * This event is triggered after archiving.
-         *
-         * @param CronArchive $this
-         */
-        Piwik::postEvent('CronArchive.end', array($this));
-
-        if (empty($this->errors)) {
-            // No error -> Logs the successful script execution until completion
-            Option::set(self::OPTION_ARCHIVING_FINISHED_TS, time());
-            return;
-        }
-
-        $this->logSection("SUMMARY OF ERRORS");
-        foreach ($this->errors as $error) {
-            // do not logError since errors are already in stderr
-            $this->logger->info("Error: " . $error);
+        foreach ($archiveArray as $entry) {
+            if ($entry['idsite'] == $archive['idsite']
+                && $entry['period'] == $archive['period']
+                && $entry['date1'] == $archive['date1']
+                && $entry['date2'] == $archive['date2']
+                && $entry['name'] == $archive['name']
+            ) {
+                return true;
+            }
         }
-
-        $summary = count($this->errors) . " total errors during this script execution, please investigate and try and fix these errors.";
-        $this->logFatalError($summary);
+        return false;
     }
 
-    public function logFatalError($m)
+    private function getNextInvalidatedArchive($periodToGet)
     {
-        $this->logError($m);
+        $tables = $this->getTablesWithInvalidatedArchives();
 
-        throw new Exception($m);
+        foreach ($tables as $table) {
+            $tableMonth = substr($table, strlen($table) - 7, 7);
+            $tableMonth = str_replace('_', '-', $tableMonth);
+
+            if (!isset($this->idArchivesToExclude[$tableMonth])) {
+                $this->idArchivesToExclude[$tableMonth] = [];
+            }
+
+            $nextArchive = $this->model->getNextInvalidatedArchive($table, $periodToGet, $this->allWebsites, $this->idArchivesToExclude[$tableMonth]);
+            if (!empty($nextArchive)) {
+                $this->findSegmentForArchive($nextArchive);
```

Comment:
This is done in `SegmentArchivingRequestUrlProvider`

Reply:
@diosmosis the code for that class is gone or maybe I'm looking wrong?

Reply:
@tsteur sorry, I renamed the class to `SegmentArchiving`

Reply:
@diosmosis not sure I understand. This class is not used here at this point? or is this supposed to work because `getSegmentArchivesToInvalidateForNewSegments ()` will only ever invalidate reports for segments with auto archiving enabled? Not sure if other segments would be somehow invalidated as well? Might need to check it here again?

Reply:
> or is this supposed to work because getSegmentArchivesToInvalidateForNewSegments () will only ever invalidate reports for segments with auto archiving enabled? 

It should only invalidate reports if `auto_archive` is set to `1` currently.

Reply:
So it needs to be added here as well right? Cause otherwise if someone invalidates reports, they would suddenly be archived through cron when they actually should only be archived through browser archiving.

Reply:
Oh yes, it should, will add 👍